pkg/handlers/httpproxy/inject.go

```diff
@@ -40,14 +41,21 @@ const (
 type httpProxyPatchHandler struct {
 	decoder runtime.Decoder
 	client  ctrlclient.Reader
+
+	variableName      string
+	variableFieldPath []string
 }
 
 var (
 	_ handlers.Named           = &httpProxyPatchHandler{}
 	_ mutation.GeneratePatches = &httpProxyPatchHandler{}
 )
 
-func NewPatch(cl ctrlclient.Reader) *httpProxyPatchHandler {
+func NewPatch(
+	cl ctrlclient.Reader,
+	variableName string,
+	variableFieldPath ...string,
```

Comment:
This is repeated between handlers. Can't the meta generate patch figure out how to map variables?

Reply:
Possibly - any ideas how?

Reply:
Some code needs to be aware about the relationship between our the `ClusterConfig` variable and partial variables placement and the handler that is interested in the variable. I can only thing about creating additional code where we'd capture this mapping and I am not sure if it would be any better compare to handler knowing where to reach for given variable.